scripts/extract_tra_text.py

```diff
@@ -1,40 +1,45 @@
 #!/usr/bin/env python3
-import json
-from pathlib import Path
-
+from models.mod import MetaStatusEnum
 from scripts.utils import ModManager
 from settings import LANGUAGE_DEFAULT
 
+DATA_SEP = ";"
+
 
 def extract_mods_text(language):
-    root = Path.cwd()
-    filename = ModManager.db_path() / ModManager.get_language_filename(language=language)
+    """
+    Extract the text of the mods to translate from the database.
 
-    # use root in paths
-    with open(filename, "r", encoding="utf-8") as f:
-        mods = json.load(f)
+    Outputs:
+        - tra_input_{language}.txt: the text of the description and notes to translate
+        - tra_input_map_{language}.txt: the map of the text to the mod id, text type and index
+    """
+    mods = ModManager.load(language=language)
 
+    db_path = ModManager.db_path()
     with (
-        open(root / "db" / f"tra_input_{language}.txt", "w", encoding="utf-8") as out,
-        open(root / "db" / f"tra_input_map_{language}.txt", "w", encoding="utf-8") as map_out,
+        open(db_path / f"tra_input_{language}.txt", "w", encoding="utf-8") as out,
+        open(db_path / f"tra_input_map_{language}.txt", "w", encoding="utf-8") as map_out,
     ):
         line_number = 1
 
         for mod in mods:
             mod_id = mod["id"]
 
-            if mod["description_meta"]["status"] == "todo":
-                # get description_meta.source
+            if mod["description_meta"]["status"] == MetaStatusEnum.TODO:
                 source = mod["description_meta"]["source"]
                 out.write(source + "\n")
-                map_out.write(f"{line_number}: mod {mod_id} description\n")
+                map_line = DATA_SEP.join([str(line_number), str(mod_id), "description", "0"])
+                map_out.write(map_line + "\n")
                 line_number += 1
 
-            # Write each note on a new line, check if note list is not empty and note is not empty string
-            if mod["notes_meta"]["status"] == "todo":
+            if mod["notes_meta"]["status"] == MetaStatusEnum.TODO:
                 for note_idx, note in enumerate(mod["notes_meta"]["source"]):
                     out.write(note + "\n")
-                    map_out.write(f"{line_number}: mod {mod_id} note {note_idx}\n")
+                    map_line = DATA_SEP.join(
+                        [str(line_number), str(mod_id), "note", str(note_idx)]
+                    )
+                    map_out.write(map_line + "\n")
```

Comment:
Since the logic is specific and present twice, it might be better to extract it in a separate method.

```py
def build_line(line_number, mod_id, field_type, idx):
    return DATA_SEP.join([str(line_number), str(mod_id), field_type, str(idx)])
```
And if it contains the logic for building a line, why doesn't it contain the logic for the parsing ?

```py
def parse_line(line):
   line_number, mod_id, field_type, idx = line.split(DATA_SEP)
   return int(line_number), int(mod_id), field_type, int(idx)
```
It can be useful in merge_tra_text.

Reply:
Good pint! Will do these, should I use type hints as well, or skip them?

Reply:
It's better with typing!

Reply:
Done, added the two with typing in https://github.com/RiwsPy/lcc-docs/pull/13/commits/ecce51c36bafd866e64910b92030804d5b320a80!